packages/react-google-maps-api/src/__tests__/utils/injectscript.test.ts

```diff
@@ -0,0 +1,143 @@
+import { injectScript } from '../../utils/injectscript'
+
+interface WindowWithGoogleMap extends Window {
+  initMap?: () => void;
+}
+
+// interface ScriptWithOnErrorHandler extends Element {
+//     onerror?: (err: Error) => void;
+// }
+
+describe('utils/injectScript', () => {
+  // let urls: string[] = [];
+
+  // function createScriptUrl(script = '(function(){ var a = 1; })()') {
+  //     const b = new Blob([script], { type: 'text/javascript' });
+  //     const url = URL.createObjectURL(b);
+  //     urls.push(url)
+  //     return url;
+  // }
+
+  // afterAll(() => urls.forEach(url => URL.revokeObjectURL(url)));
+
+  describe('injectScript', () => {
+    it('adds a script to the page with id and src attributes that correspond to the id and url properties', () => {
+      const url = `http://localhost/script1.js`
+      const value = injectScript({
+        id: 'injectScript-test1',
+        url,
+      })
+
+
+      setTimeout(() => {
+        const win: WindowWithGoogleMap = window
+        if(win.initMap) {
+          win.initMap()
+        }
+      }, 0)
+
+      const element = document.querySelector('#injectScript-test1')
+      expect(element).toBeTruthy()
+      expect(element).toHaveProperty('src', url)
+      return value
+    })
+
+    it('returns a promise which resolves when the script loads', () => {
+      const id = 'injectScript-test2'
+      const url = `http://localhost/script2.js`
+
+      setTimeout(() => {
+        const win: WindowWithGoogleMap = window
+        if(win.initMap) {
+          win.initMap()
+        }
+      }, 0)
+
+      return injectScript({
+        id,
+        url,
+      })
+        .then(value => expect(value).toBeTruthy())
+    })
+
+
+    // it('returns a promise which rejects if the script does not load', () => {
```

Comment:
I wonder why this test doesn't work. Is it some technical limitation?

Reply:
I think it is a limitation with jsdom. It allows setting an onerror handler for the script but always returned a null value when you tried to retrieve it.